### PR TITLE
Bug 1062787 - Use the new shared helper for loading JSON files in the sm...

### PR DESCRIPTION
**Pull request by Gioyik**
...art collections code

### CODE REVIEW

**Comment by try-server-hook** [comment]
[Gioyik Giovanny Andres Gongora Granada (Gioyik)](https://github.com/Gioyik) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8148434276f9)

**Comment by try-server-hook** [comment]
[Gioyik Giovanny Andres Gongora Granada (Gioyik)](https://github.com/Gioyik) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8e7164dec50d)

**Comment by try-server-hook** [comment]
[Gioyik Giovanny Andres Gongora Granada (Gioyik)](https://github.com/Gioyik) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f6ab54f09420)

**Comment by try-server-hook** [comment]
[Gioyik Giovanny Andres Gongora Granada (Gioyik)](https://github.com/Gioyik) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=c034bd8f7b81)

**Comment by try-server-hook** [comment]
[Gioyik Giovanny Andres Gongora Granada (Gioyik)](https://github.com/Gioyik) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=1b84dc744ce2)

File: apps/collection/js/setup.js [hunk 1]
```diff
@@ -1,11 +1,9 @@
 'use strict';
 
-/* global CollectionsDatabase */
+/* global CollectionsDatabase, LazyLoader */
 
 (function(exports) {
 
-  const PRE_INSTALLED_COLLECTIONS_FILE = 'js/pre_installed_collections.json';
```
Comment (gabrielesvelto) [inline apps/collection/js/setup.js]: Do not remove this declaration, use it instead in the code below like the existing code does.

File: apps/collection/js/setup.js [hunk 2]
```diff
@@ -32,28 +30,12 @@
...
-      }.bind(this);
-
-      xhr.onerror = function _xhrOnError() {
-        this.onError('file not found');
-      }.bind(this);
-
-      xhr.send();
+      LazyLoader.getJSON('js/pre_installed_collections.json')
+        .then(function(json) {
+          this.populate(JSON.parse(json));
```
Comment (gabrielesvelto) [inline apps/collection/js/setup.js]: `this` does not refer to the correct object here. You need to bind the function appropriately like the existing ones did for it to work.

File: apps/collection/js/setup.js [hunk 3]
```diff
@@ -32,28 +30,12 @@
...
+        }, function(error) {
+          console.log('file not found');
```
Comment (gabrielesvelto) [inline apps/collection/js/setup.js]: Note that the existing code used the `onError` function, not `console.log()`, you should do the same.

**Comment by try-server-hook** [comment]
[Gioyik Giovanny Andres Gongora Granada (Gioyik)](https://github.com/Gioyik) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=34bb59386f30)

File: apps/collection/js/setup.js [hunk 4]
```diff
@@ -32,28 +32,16 @@
...
-      }.bind(this);
-
-      xhr.send();
+      LazyLoader.getJSON(PRE_INSTALLED_COLLECTIONS_FILE)
+        .then(function(json) {
+          try {
+            this.populate(JSON.parse(json));
+          } catch (ex) {
+            this.onError(ex);
+          }
```
Comment (gabrielesvelto) [inline apps/collection/js/setup.js]: As mentioned before this won't work as the function isn't bound to the correct `this` reference. Look at the existing code and what it does.

File: apps/collection/js/setup.js [hunk 5]
```diff
@@ -32,28 +32,16 @@
...
+        }, function(error) {
+          this.onError('file not found');
+        });
```
Comment (gabrielesvelto) [inline apps/collection/js/setup.js]: Likewise.

**Pull request closed without merging**